src/http.cc

```diff
@@ -1933,6 +2064,45 @@ HttpStateData::httpBuildRequestHeader(HttpRequest * request,
     strConnection.clean();
 }
 
+static void
+makeUpgradeHeader(HttpStateData *state, HttpRequest *req, const HttpHeader &hdr_in, HttpHeader &hdr_out)
+{
+    if (!state)
+        return;
+
+    if (!hdr_in.has(Http::HdrType::UPGRADE))
+        return;
+
+    const String upgradeIn = hdr_in.getList(Http::HdrType::UPGRADE);
+    String upgradeOut;
+    const char *pos = nullptr;
+    const char *item;
+    int ilen = 0;
+    while (strListGetItem(&upgradeIn, ',', &item, &ilen, &pos)) {
+        //Config.accessList.http_upgrade_protocols
+        SBuf proto(item, ilen);
+        auto it = Config.accessList.http_upgrade_protocols->find(proto);
+
+        // If protocol not found try the "ANY" protocol
+        if (it == Config.accessList.http_upgrade_protocols->end())
+            it = Config.accessList.http_upgrade_protocols->find(SBuf("ANY"));
```

Comment:
Please optimize, creating a static SBuf for "ANY".

Reply:
Yes, please.

Reply:
> Here is why I am still thinking that we need to correct this documentation. Before your changes, the code simply scheduled HttpStateData::processReply() call of the existing HttpStateData object (thus resuming processing)
> Now, for 101 response the new code drops HttpStateData object instead (as you noted in another comment, thus not "resuming" processing (with the existing HttpStateData object).

Yes this is true. However the HttpStateData has stopped any processing and waiting for the proceedAfter1XXX call to "restore states and resume processing after 1xxx is forwarded". The processing may result to finish its job yes.

> Please optimize, creating a static SBuf for "ANY".

OK.